keyserver/listener.go

```diff
@@ -100,6 +100,37 @@ func (server *ConiksServer) handleClientMessage(msg []byte) ([]byte, error) {
 
 	// handle request
 	switch req.Type {
+	default:
+		log.Printf("unknown message type: %q", req.Type)
+		response = p.NewErrorResponse(p.ErrorMalformedClientMessage)
+		err = p.Error(p.ErrorMalformedClientMessage)
+	}
+
+marshalling:
+	res, e := json.Marshal(response)
+	if e != nil {
+		panic(p.Error(p.ErrorInternalServer))
+	}
+	return res, err
+}
+
+func (server *ConiksServer) handleBotMessage(msg []byte) ([]byte, error) {
```

Comment:
It seems like ``handleClientMessage()`` shares a fair amount of code with ``handleBotMessage()``. Maybe it's possible to refactor some of this repeated code into small functions? For example, L121-L130 is reused, and the default case of the switch is identical in both functions.